dont set state after component unmounting

clif-os opened the pull request:
Wanted to use https://developer.mozilla.org/en-US/docs/Web/API/AbortController#Browser_compatibility, but it is not supported in IE.

Using old way of avoiding setting state after component unmount:
https://codesandbox.io/s/nwwrrpvqw4?file=/src/axios-track-mounted.js

(verified demo and tests still function properly)

superhighfives commented:
This looks great, @clif-os—one thought, though: should we set a limit on retrying? Let's say after 3 or 5 tries, throw an error saying "Could not be fetched"?

🙌

clif-os commented:
@superhighfives that's a good idea. i think fetchTries can also just be a class var and not in state.

clif-os commented:
@superhighfives I added the bail after 5 retries. I also realized that this timeout length was resolving to Infinity and fixed that as well:
https://github.com/heroku/react-malibu/blob/master/src/MalibuSprites.js#L56

I also moved the retry count out of state since it doesnt affect anything in `render`

Tested it out by manually throwing an error and it was working as expected:

![Screen Shot 2020-07-02 at 10 28 38 AM](https://user-images.githubusercontent.com/13678764/86396369-a6cec700-bc56-11ea-9804-bd10b63a34b5.png)

superhighfives commented:
This is great—thanks @clif-os!

superhighfives commented:
This is now live @clif-os—https://www.npmjs.com/package/@heroku/react-malibu v4.2.0
